src/get-serve-d.ts

```diff
@@ -0,0 +1,43 @@
+import downloadRelease from "@terascope/fetch-github-release"
+import { join, dirname, extname } from "path"
+import { remove, ensureDir } from "fs-extra"
+import decompress from "decompress"
+
+const assetMap = {
+  win32: "windows",
+  darwin: "osx",
+  linux: "linux",
+}
+
+// function to download serve-d binaries from GitHub
+export async function getServeD() {
+  const distFolder = join(dirname(__dirname), "dist")
+  await remove(distFolder)
+  await ensureDir(distFolder)
+
+  const platform = assetMap[process.platform]
+
+  const decompressPlugins = []
+  if (process.platform !== "win32") {
+    decompressPlugins.push(await import("decompress-tarxz"))
+  }
+
+  const assets = ((await downloadRelease(
+    /* username */ "Pure-D",
+    /* repo */ "serve-d",
+    /* download folder */ distFolder,
+    /* filter release */ undefined,
+    /* filter asset */ (asset) => asset.name.indexOf(platform) >= 0,
```

Comment:
this test should probably be a little more exact

you might want to copy the function from https://github.com/Pure-D/code-d/blob/master/src/installer.ts#L261

Reply:
But do you build for other architectures? If you don't there is no need to add code for it.

Reply:
I don't build for other architectures yet, but vscode is supported on ARM, so if there is an ARM user (e.g. coders on cheap linux ARM laptops or raspberry pi) they don't download a release that is unsupported anyway.

It probably makes sense to move all the release downloading & manual compilation into an NPM package which can be used by all JS based extensions.